selfdrive/hardware/tici/agnos.py

```diff
@@ -38,30 +39,18 @@ def read(self, length):
     self.sha256.update(result)
     return result
 
-
-def unsparsify(f):
+def unsparsify(f: StreamingDecompressor) -> Generator[bytes, None, None]:
   magic = struct.unpack("I", f.read(4))[0]
   assert(magic == 0xed26ff3a)
 
   # Version
-  major = struct.unpack("H", f.read(2))[0]
```

Comment:
Don't think this makes it more clear. Currently all lines match the spec: https://source.android.com/devices/bootloader/images#sparse-format. Feel free to add comment linking to the spec.

Reply:
Ah yea this puts it into perspective more, it was literally 1:1 with the spec lol.

The stuff that runs once doesn't really matter then, but depending on how many times this loop runs though, it might be worth keeping the changes in there and even using [struct.Struct](https://docs.python.org/3/library/struct.html#struct.Struct) since calling `struct.pack` repeatedly recompiles this object in the background each time.. I'll get to work :P